quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -152,18 +153,35 @@ impl TypedSourceFactory for FileSourceFactory {
             let (dir_uri, file_name) = dir_and_filename(filepath)?;
             let storage = ctx.storage_resolver.resolve(&dir_uri).await?;
             let file_size = storage.file_num_bytes(file_name).await?.try_into().unwrap();
-            storage
-                .get_slice_stream(
-                    file_name,
-                    Range {
-                        start: offset,
-                        end: file_size,
-                    },
-                )
-                .await?
+            // If it's a gzip file, we can't seek to a specific offset, we need to start from the
+            // beginning of the file, decompress and skip the first `offset` bytes.
+            if filepath.extension().map_or(false, |ext| ext == "gz") {
```

Comment:
nitpick. not even sure if that works, but would
```suggestion
            if filepath.extension() == Some("gz") {
```